src/components/Main/Members.tsx

```diff
@@ -1,3 +1,354 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { v4 as uuid } from 'uuid';
 
-export const Members = () => <>You a members</>;
+// material UI
+import { fade, makeStyles, withStyles, Theme, createStyles } from '@material-ui/core/styles';
+import List from '@material-ui/core/List';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import ListSubheader from '@material-ui/core/ListSubheader';
+import ListItemText from '@material-ui/core/ListItemText';
+import Divider from '@material-ui/core/Divider';
+import Grid from '@material-ui/core/Grid';
+import Paper from '@material-ui/core/Paper';
+import Table from '@material-ui/core/Table';
+import TableBody from '@material-ui/core/TableBody';
+import TableCell from '@material-ui/core/TableCell';
+import TableContainer from '@material-ui/core/TableContainer';
+import TableHead from '@material-ui/core/TableHead';
+import TableRow from '@material-ui/core/TableRow';
+import SearchIcon from '@material-ui/icons/Search';
+import AddIcon from '@material-ui/icons/Add';
+import InputBase from '@material-ui/core/InputBase';
+import ButtonGroup from '@material-ui/core/ButtonGroup';
+import Button from '@material-ui/core/Button';
+import IconButton from '@material-ui/core/IconButton';
+import { green, red, amber } from '@material-ui/core/colors';
+import AddCircleIcon from '@material-ui/icons/AddCircle';
+import RemoveCircleIcon from '@material-ui/icons/RemoveCircle';
+import CSS from 'csstype';
+
+// other components
+import { ConfirmDialog } from '../shared/ConfirmDialogue';
+import { AddUserDialog } from '../shared/AddUserDialogue';
+import { AddRoleDialog } from '../shared/AddRoleDialogue';
+
+// actions
+import { getAllUsers } from '../../store/apis';
+
+// dummy data
+import {userData} from './membersDatabase';
+
+// types
+import {UserType} from '../../shared/types/membersModel';
+
+var rows = userData;
+
+const StyledTableCell = withStyles((theme: Theme) =>
+  createStyles({
+    head: {
+      backgroundColor: theme.palette.common.black,
+      color: theme.palette.common.white,
+    },
+    body: {
+      fontSize: 14,
+    },
+  }),
+)(TableCell);
+
+const styleHead: CSS.Properties = {
+  fontWeight: 'bold'
+}
+
+export const Members = () => {
+  const classes = useStyles();
+  const [database, setDatabase] = useState<UserType[]>(userData);
+  const [selected, setSelected] = useState<string[]>([]);
+  const [searchfield, setSearchField] = useState<string>('');
+  const [openConfirm, setOpenConfirm] = React.useState(false);
+  const [openAdd, setOpenAdd] = React.useState(false);
+  const [selectedValue, setSelectedValue] = React.useState(false);
+  const [selectionExists, setSelectionExists] = React.useState(true);
```

Comment:
we can work on the naming after your next pull request, since i'm sure you're going to change a lot of these